### PR TITLE
Api keys restricted

### PR DESCRIPTION
**Pull request by MariaCheca**
- Restricted access to API keys management for users with engine enabled.
- Added CTA for API keys.
- Added button 'Get more space' in mobile apps view, waiting for @brittanymicek to give a better option for the text.

screen shot 2016-10-18 at 15 22 43
screen shot 2016-10-18 at 15 24 04

CR frontend @javierarce CR backend @javitonino 
Design review all yours @josecruz :)

Fixes https://github.com/CartoDB/cartodb-management/issues/4646.

### CODE REVIEW

File: app/controllers/admin/client_applications_controller.rb [hunk 1]
```diff
@@ -55,4 +56,11 @@ def regenerate_oauth
     redirect_to CartoDB.url(self, 'oauth_credentials', {type: 'oauth'}, current_user),
                 :flash => {:success => "Your OAuth credentials have been updated successfully"}
   end
+
+  private
+  def engine_enabled?
```
Comment (javitonino) [inline app/controllers/admin/client_applications_controller.rb]: Method names ending in `?` should return true/false. As this does not (it renders on error), rename this to something like `enforce_engine_enabled`, `engine_users_only` or something along this lines.

Also, we usually leave a blank line after `private`, but that's just cosmetic.

File: app/controllers/admin/organizations_controller.rb [hunk 2]
```diff
@@ -121,4 +122,10 @@ def load_ldap_configuration
     @ldap_configuration = Carto::Ldap::Configuration.where(organization_id: @organization.id).first
   end
 
+  def engine_enabled?
```
Comment (javitonino) [inline app/controllers/admin/organizations_controller.rb]: Same here.

File: app/views/carto/admin/mobile_apps/_mobile_plan_info.html.erb [hunk 3]
```diff
@@ -8,6 +8,8 @@
       <div class="MobilePlan-subtitle"><%= 'Xamarin features enabled. ' if current_user.mobile_xamarin %><%= 'Custom watermark. ' if current_user.mobile_custom_watermark %><%= 'Offline maps. ' if current_user.mobile_offline_maps %><%= 'GIS extension.' if current_user.mobile_gis_extension %></div>
     </div>
   </div>
+
+  <a href="<%= current_user.upgrade_url(request.protocol) %>" class="CDB-Text CDB-Size-small is-semibold CDB-Button CDB-Button--secondary u-upperCase">Get more space</a>
```
Comment (javitonino) [inline app/views/carto/admin/mobile_apps/_mobile_plan_info.html.erb]: I think "get more space" sounds strange here, as mobile SDK limits by users, not space.

File: app/controllers/admin/client_applications_controller.rb (hunk 1 above)
Comment (javitonino) [inline app/controllers/admin/client_applications_controller.rb]: Also, I'd return a 403, as the page is there but is forbidden for this user.

**Comment by javierarce** [comment]
👍  too

**Comment by brittanymicek** [comment]
@MariaCheca 

Please check that the capitalization of "keys" in "API keys" is consistent with the other pages.

Change "Get More Space" to [OPTIONS!!!!]
"Grant More Users"
"Extend Limits"
"Level up"
"Expand Quota"

**Comment by MariaCheca** [comment]
Hey @brittanymicek, just one more thing, we've seen this tip message at the end of the API keys form section, and we're wondering if the text is right:
screen shot 2016-10-19 at 11 20 18

Shouldn't it be "Learn more about using your simple **API key** in CARTO"? It leads to the documentation page about API keys (https://carto.com/docs/carto-editor/your-account/#api-key).

Thank you!

**Comment by xavijam** [comment]
Hey @brittanymicek, could you help us with that? Pretty please 🤗 .

**Comment by brittanymicek** [comment]
@MariaCheca 
"Learn more about using your API key in CARTO"

**Comment by xavijam** [comment]
So, pending things:
- [ ] Change text (thanks @brittanymicek)
- [ ] Remove "App keys" link from user dropdown if engine is not enabled (remember to do it in central after this PR).

**Pull request merged**